### PR TITLE
Feature: add Per Form Gateways add-on to the form migration process

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-60] [GIVE-184]

Related PR: https://github.com/impress-org/give-per-form-gateways/pull/11

## Description

This PR adds the necessary logic to transfer data from the Per Form Gateways addon settings to the migrated forms.

This PR also sets the Per Form Gateways addon as supported by the visual form builder.

## Affects

The migrated forms.

## Visuals

image

## Testing Instructions

1. Install the Per Form Gatways addon from this PR/branch: https://github.com/impress-org/give-per-form-gateways/pull/11
2. Create a legacy form and select a few gateways to be displayed when the form gets rendered
3. Do the migration process, then check the Payment Gateways block settings in the migrated form, you should see in the block settings the gateways selected in the previous step, except the ones not compatible with v3 forms.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-60]: https://stellarwp.atlassian.net/browse/GIVE-60?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-184]: https://stellarwp.atlassian.net/browse/GIVE-184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/FormMigration/Steps/PaymentGateways.php [hunk 1]
```diff
@@ -10,6 +10,7 @@
 class PaymentGateways extends FormMigrationStep
 {
     /**
+     * @unreleased Add Per Form Gateways settings to this step
```
Comment (kjohnson) [inline src/FormMigration/Steps/PaymentGateways.php:13]: I would suggest adding a new migration step for the Per Form Gateway settings.

**Review by kjohnson — CHANGES_REQUESTED** [review]

File: src/FormMigration/Steps/PaymentGateways.php (hunk 1 above)
Comment (glaubersilva) [reply src/FormMigration/Steps/PaymentGateways.php]: Resolved in c99e07a84ef766f41f831adf7b930b60e75e893b

**Comment by glaubersilva** [comment]
@kjohnson This is ready for re-review.

**Review by kjohnson — APPROVED** [review]
Thanks for splitting the migration logic into a separate step.

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**